integration/container/run_test.go

```diff
@@ -0,0 +1,44 @@
+package container // import "github.com/docker/docker/integration/container"
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/versions"
+	"github.com/docker/docker/integration/internal/container"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/poll"
+	"gotest.tools/v3/skip"
+)
+
+func TestConsoleSize(t *testing.T) {
+	skip.If(t, versions.LessThan(testEnv.DaemonAPIVersion(), "1.42"), "skip test from new feature")
+
+	defer setupTest(t)()
+	client := testEnv.APIClient()
+	ctx := context.Background()
+
+	cID := container.Run(ctx, t, client,
+		container.WithTty(true),
+		container.WithImage("debian:bullseye-slim"),
+		container.WithCmd("sh", "-c", "tput lines; tput cols"),
+		container.WithConsoleSize(57, 123),
+	)
+
+	poll.WaitOn(t, container.IsInState(ctx, client, cID, "exited"), poll.WithDelay(100*time.Millisecond))
+
+	out, err := client.ContainerLogs(ctx, cID, types.ContainerLogsOptions{ShowStdout: true})
+	assert.NilError(t, err)
+	defer out.Close()
```

Comment:
It's a bit unfortunate that we currently don't have a way to return the output from `container.Run()` (as we have for `container.Exec()`). Perhaps something we should look into (we could have a `WithOutputBuffer()` (e.g.) function that we can pass a buffer).

Also not something for this PR though; just thinking out loud here 😅